artichoke-core/src/module_registry.rs

```diff
@@ -0,0 +1,41 @@
+use std::any::Any;
+
+pub trait ModuleRegistry {
+    /// Module value type
+    type Value;
+
+    /// Module error type
+    type Error;
+
+    /// Module specification type
+    type Spec;
```

Comment:
What do you think of these docs?

```suggestion
    /// Concrete value type for boxed Ruby values.
    type Value;

    /// Concrete error type for errors encountered when manipulating the module registry.
    type Error;

    /// Type representing a module specification.
    type Spec;
```